### PR TITLE
fix: search page meta

### LINKED ISSUE #38439: /news search results page doesn't produce Twitter card
Currently the search results page of /news doesn't have the necessary meta properties to produce a large Twitter card. I think we need an `og:image` and twitter image property for this. We also need the proper tags so that Twitter can display a summary. I imagine this is also the case with Facebook and LinkedIn. Here's an example of the card not rendering on Twitter:

https://twitter.com/ossia/status/1242844586357161987

### PR DESCRIPTION
**Pull request by scissorsneedfoodtoo**
This PR treats the search page as a fully static page rather than inserting it into `default.hbs` like we did before.

With the old method where we inserted everything into `default.hbs`, we would have had to create a search results page in Ghost to properly use the `{{ghost_head}}` helper like we were before. Even then, we would have to manually add the meta title and description to the search results page in Ghost. We would also have to update `routes.yml` to pick up the data from that page.

I figured this would be a bit easier since it's basically a static page with content that's loaded from the Algolia API.

This PR also removes the unused `partials/icons/search.hbs` file and updates `partials/algolia-search.hbs` to use https instead of http.

Here's a screenshot of the card on Twitter:

image

And Facebook:

image

Cc @ahmadabdolsaheb.

Closes: https://github.com/freeCodeCamp/freeCodeCamp/issues/38439

### CODE REVIEW

**Comment by darek-grund** [comment]
What are benefits of creating static page instead of letting it handle by Ghost? I see additional risks, for example:
- from now on you would need to manually recreate search page every time something changes in partials, most importantly if page head will change,
- image url are hardcoded, what if someone uploads new images for social media? You would need to remember to update them in search page too,
- new css added to site - you would need to match that on search page,
- new js is added to site - you would need to match that on search page

Basically, every change would require contributors to think "do I need to update search page too? May it break anything?". I would strongly advise to share common parts between templates, not to create separate html pages, because it will be hard to maintain such site.

**Comment by scissorsneedfoodtoo** [comment]
Hi @darek-grund, thanks for asking these questions and for the details you provided. It will really help us make an informed decision going forward.

We use dynamic routing for the search page, and don't have an actual page with the slug `/search`. I believe we did this to prevent someone from accidentally deleting the search page and breaking that functionality.

While I definitely agree that using templates is better whenever possible, I wasn't able to get the headers to play nicely with a dynamically routed page. There were a few people who had the same issue on the Ghost forums, and I wasn't able to find a great solution.

Using the current code / template, I had to publish a search page in Ghost, add the meta title and description through the UI, upload the correct picture, and modify `routes.yaml` to get the cards working the way we want.

CSS and some other things are still pulled using handlebars helpers. But you're right that we would need manually add scripts or styles if new ones are added and we want to include them on the search page. 

For pictures and meta information, we would either need to update search page through Ghost, or by submitting a PR to this repo.

**Comment by darek-grund** [comment]
@scissorsneedfoodtoo thank you for detailed explanation - this helps me a lot, because I was actually wondering how the search page was done in production.

> Using the current code / template, I had to publish a search page in Ghost, add the meta title and description through the UI, upload the correct picture, and modify routes.yaml to get the cards working the way we want.

That's exactly how it should be done. And you also need to create template for search page and assign it to that page. Then everything should work out of the box. But I'm wondering what issues with header did you encounter with that approach? Maybe there's something I'm not aware of.

**Comment by scissorsneedfoodtoo** [comment]
Of course, and thank you again for all the details you provided @darek-grund.

I'm fine with doing things that way, especially if that's the usual way to do things in Ghost. There's still the possibility that someone could accidentally remove the search page from Ghost and break that functionality, but it may be a non-issue soon.

The thing I found in my testing on .dev/news is that, when you create a search page on Ghost and use it as a template in the `routes.yaml` file, you still need to set the cover image, and meta title and description on the search page itself in the Ghost UI. It doesn't seem to pick up the defaults from Ghost itself. If you leave those things blank, you'll get a blank card like the one Quincy posted in the issue.

That might be default behavior, though. Definitely makes sense that posts are handled that way since the cover images and meta information should unique for each post.

Because of that behavior, it seems like our options are basically to handle future updates through PRs like this, or through the Ghost UI itself. Either way is totally fine with me, though.

File: search-results.hbs [hunk 1]
```diff
@@ -1,154 +1,215 @@
...
+    <meta name="twitter:image" content="https://www.freecodecamp.org/news/content/images/2019/11/fcc_ghost_publication_cover.png" />
+    <meta name="twitter:card" content="summary_large_image" />
+    <meta name="twitter:site" content="@tryghost" />
+
+    <meta property="og:title" content="Search Results" />
+    <meta property="og:description" content="Learn to code at home. Build projects. Earn certifications. 
+    Since 2014, more than 40,000 freeCodeCamp.org graduates have gotten jobs at tech companies including 
+    Google, Apple, Amazon, and Microsoft." />
+    <meta property="og:url" content="https://www.freecodecamp.org/news/search/" />
+    <meta property="og:image" content="https://www.freecodecamp.org/news/content/images/2019/11/fcc_ghost_publication_cover.png" />
```
Comment (ahmaxed) [inline search-results.hbs]: Should we have these in a CDN similar to learn?
It seems that this link would break if the original file is removed or modified.

Comment (scissorsneedfoodtoo) [reply search-results.hbs]: If I remember correctly, I got this from the publication cover section of the Ghost settings page. I could make use a CDN link if that's better.

Comment (ahmaxed) [reply search-results.hbs]: yes, I believe that is how we are handling them on /learn

Comment (scissorsneedfoodtoo) [reply search-results.hbs]: Sure thing, I can do that and update this PR.

**Comment by scissorsneedfoodtoo** [comment]
Will serve the image from our CDN rather than Ghost itself. This requires https://github.com/freeCodeCamp/cdn/pull/52 to be merged first.
